### PR TITLE
add buildx to github action

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -221,3 +221,20 @@ jobs:
...
+        uses: actions/checkout@v2
+      - uses: docker/setup-buildx-action@v1
+        id: buildx
+        with:
+          install: true
+      - name: Build
+        run: |
+          docker build -f /home/runner/work/${{ github.repository }}/build/cloud/Dockerfile # will run buildx
+        with:
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
```
Comment (fisherxu) [inline .github/workflows/main.yaml:240]: Should do the make command from makefile.

Comment (gy95) [reply .github/workflows/main.yaml]: got it, i'll integrate with buildx in https://github.com/kubeedge/kubeedge/pull/3530

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -221,3 +221,20 @@ jobs:
           fetch-depth: 0
 
       - run: make image
+
+  buildx:
```
Comment (zhu733756) [inline .github/workflows/main.yaml:225]: Seems good. Have 3 questions to ask.
1 Does this work if the users don't have push permission for the image?  Do we need a script for docker logining?
2 If this CI is too slow to build and may affect other CI actions, why not move it out? 
3 Does this CI need to be daily built?  If so, we need to add a schedule for it.

**Pull request closed without merging**